controllers/fenceagentsremediation_controller_test.go

```diff
@@ -252,6 +312,32 @@ func createRunningPod(containerName, podName, nodeName string) *corev1.Pod {
 	return pod
 }
 
+func createTerminatingPod(containerName, podName, nodeName string) *corev1.Pod {
+	pod := buildPod(containerName, podName, nodeName)
+	now := metav1.Now()
+	pod.ObjectMeta = metav1.ObjectMeta{Name: pod.Name, Namespace: pod.Namespace, DeletionTimestamp: &now}
+	ExpectWithOffset(1, k8sClient.Create(context.Background(), pod)).To(Succeed())
+	return pod
+}
+
+func deleteTerminatingPod(podName string) {
```

Comment:
I think function `deleteTerminatingPod`` could be minimized, since there is no need to create a new pod based on its name in order to delete it.

In line 235 we can assign the pod to a variable (`podTerminate1 := createTerminatingPod("far-terminate-1", testPodName, workerNode)`) and then in line 253 simply call `deleteTerminatingPod(podTerminate1)` to crate nodeKey, delete the inputed pod and verify that it is gone.